cli/flox-rust-sdk/src/models/environment/core_environment.rs

```diff
@@ -305,18 +427,14 @@ impl<State> CoreEnvironment<State> {
     /// It's included in the [ReadOnly] struct for ergonomic reasons
     /// and because it doesn't modify the manifest.
     pub fn build_container(
-        &mut self,
-        _flox: &Flox,
+        lockfile_path: CanonicalPath,
     ) -> Result<ContainerBuilder, CoreEnvironmentError> {
         if std::env::consts::OS != "linux" {
             return Err(CoreEnvironmentError::ContainerizeUnsupportedSystem(
                 std::env::consts::OS.to_string(),
             ));
         }
 
-        let lockfile_path = CanonicalPath::new(self.lockfile_path())
-            .map_err(CoreEnvironmentError::BadLockfilePath)?;
-
         let mut pkgdb_cmd = Command::new(Path::new(&*PKGDB_BIN));
```

Comment:
Taking `lockfile_path` as an argument forces the caller to handle locking. The caller has to handle locking because `CoreEnvironment` doesn't know if it's a Remote, Managed, or Path environment

Reply:
Adding a comment

Reply:
Wouldn't you already be forced to lock the env as not having a lockfile will result in an error? That condition could be made stronger to require an uptodate lockfile as well..

I think if this is now only dependent on the lockfile and we dont want to check and error, we might eventually just disassociate `build_container` from `CoreEnvironment` entirely.
Either as a separate provider or implemented on lockfiles instead (i know, the latter don't have track their location yet)